### PR TITLE
MODRS-2 - Work with remote storage locations

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[MODRS-2](https://issues.folio.org/browse/MODRS-2) - Work with remote storage locations

## Purpose
 If the location attribute on UI is changed to link the location to the remote one (mark as remote) https://issues.folio.org/browse/UITEN-119, then this information should be stored in database in remote-storage module.

## Approach
* created location mappings endpoint
* updated unit tests 

## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [x] Code coverage on new code is 80% or greater
   - [x] Duplications on new code is 3% or less
   - [x] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [ ] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by MikhailFokanov — APPROVED** [review]
Looks good.

File: src/main/resources/swagger.api/remote-storage.yaml [hunk 1]
```diff
@@ -150,6 +150,72 @@ paths:
         required: true
         schema:
           type: string
+          pattern: "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[1-5][a-fA-F0-9]{3}-[89abAB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$"
```
Comment (khandramai) [inline src/main/resources/swagger.api/remote-storage.yaml:153]: Is it possible to use separate trait for this?

**Comment by khandramai** [comment]
@siarhei-charniak , could you please fix code smells?

**Comment by siarhei-charniak** [comment]
> @siarhei-charniak , could you please fix code smells?

Code smells can not be removed:
* mapstruct allows multiple `@Mapping` annotations only inside `@Mapping` annotation
* `public` inside test classes can not be removed since `TestSuite` class resides out of tests package

**Review by khandramai — APPROVED** [review]

**Pull request merged**